source/common/ssl/connection_impl.cc

```diff
@@ -251,8 +251,21 @@ std::string ConnectionImpl::subjectPeerCertificate() {
   if (!cert) {
     return "";
   }
-  bssl::UniquePtr<char> buf(X509_NAME_oneline(X509_get_subject_name(cert.get()), nullptr, 0));
-  return std::string(buf.get());
+
+  // Format settings for a single line in RFC 2253 format.
+  constexpr int indent = 0;
+  constexpr unsigned long flags = XN_FLAG_RFC2253;
```

Comment:
Istio hasn't started to use the XFCC field yet. So either one is OK to us. I prefer using `XN_FLAG_RFC2253` because it's more compact, which means the XFCC header size is smaller.